systemFlake.nix

```diff
@@ -26,19 +27,21 @@
 
 let
   evalHostArgs =
-    { channelName ? "nixpkgs"
-    , modules ? []
-    , system ? defaultSystem
-    , extraArgs ? {}
+    { channelName ? defaultHostAttrs.channelName or "nixpkgs"
+    , modules ? defaultHostAttrs.modules or [ ]
+    , system ? defaultHostAttrs.system or defaultSystem
+    , extraArgs ? defaultHostAttrs.extraArgs or { }
+    , output ? defaultHostAttrs.output or "nixosConfigurations"
+    , builder ? defaultHostAttrs.builder or channels.${channelName}.input.lib.nixosSystem
     , ...
-    }: defaultHostAttrs
-      // { 
-        inherit channelName system; 
-        modules = sharedModules ++ modules;
-        extraArgs = sharedExtraArgs // extraArgs;
-      };
+    }:
+    {
+      inherit channelName system output builder;
+      modules = sharedModules ++ modules;
+      extraArgs = sharedExtraArgs // extraArgs;
+    };
```

Comment:
I don't agree with the semantics here:

`default*` and `shared*` ought to be equivalent alternatives.

Note, that the word "default" does not mandate "override" for mergables (`[]` & `{}`).